go/mysql/mysql56_gtid_set.go

```diff
@@ -35,6 +35,35 @@ func (iv interval) contains(other interval) bool {
 	return iv.start <= other.start && other.end <= iv.end
 }
 
+// overlaps returns true if any part of |other| overlaps with this interval.
+func (iv interval) overlaps(other interval) bool {
+	return other.start >= iv.start && other.start < iv.end ||
+		other.end <= iv.end && other.end > iv.start ||
+		other.start == iv.start && other.end == iv.end
```

Comment:
```suggestion
	return other.start >= iv.start && other.start <= iv.end ||
		other.end <= iv.end && other.end >= iv.start ||
		other.start == iv.start && other.end == iv.end
```

I believe these two tests cases do not match MySQL:
```sql
SELECT GTID_SUBTRACT('3E11FA47-71CA-11E1-9E33-C80AA9429562:21-57', '3E11FA47-71CA-11E1-9E33-C80AA9429562:20-21');
SELECT GTID_SUBTRACT('3E11FA47-71CA-11E1-9E33-C80AA9429562:21-57', '3E11FA47-71CA-11E1-9E33-C80AA9429562:57-58');
```

Reply:
Might not need that last and condition with the boundary changes

Reply:
Wow, impressive catch @jycor. Thank you! 🙏  Fixed in 55ce8dbabfb5f9c103eaa7a828d971cd401a73a9 and I'll add those test cases to GMS, too.